### PR TITLE
Adding Armstrong number

### PR DESCRIPTION
**Pull request by kostogls**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

File: maths/armstrong_numbers.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+""" An Armstrong number is a number that is equal to the sum of the cubes of its digits.
+For example, 370 is an Armstrong number because 3*3*3 + 7*7*7 + 0*0*0 = 370.
+An Armstrong number is often called Narcissistic number.
+"""
+
+
+def armstrong_number(n):
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: Please remove the print() statements from the algorithm itself and just have the function return a bool.  True/False

Put your print() statements in the code that calls the function.

Comment (kostogls) [reply maths/armstrong_numbers.py]: Done! :)

File: maths/armstrong_numbers.py [hunk 2]
```diff
@@ -30,15 +30,19 @@ def armstrong_number(n):
         sum = sum + (rem ** number_of_digits)
         temp //= 10
     if n == sum:
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: `return n == sum`

File: maths/armstrong_numbers.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    """
+        This function checks if a number is Armstrong or not.
+
+        >>> armstrong_number(153)
+        True
+        >>> armstrong_number(200)
+        False
+        >>> armstrong_number(1634)
+        True
+        """
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: Add three more tests:
```
>>> armstrong_number(0)
???
>>> armstrong_number(-1)
???
>>> armstrong_number(1.2)
???
```

File: maths/armstrong_numbers.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    num = int(input("Enter a number to check if it is Armstrong or not: ").strip())
+    armstrong_number(num)
+    if armstrong_number(num):
+        print(num, " is an Armstrong number")
+    else:
+        print(num, " is not an Armstrong number")
+
+
+if __name__ == '__main__':
+    main()
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: Move `main()` after the tests have passed.

File: maths/armstrong_numbers.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    temp = n
+    # Calculation of digits of the number
+    while temp > 0:
+        number_of_digits += 1
+        temp //= 10
+    # Dividing number into separate digits and find Armstrong number
+    temp = n
+    while temp > 0:
+        rem = temp % 10
+        sum = sum + (rem ** number_of_digits)
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: sum += (rem ** number_of_digits)

File: maths/armstrong_numbers.py [hunk 6]
```diff
@@ -30,15 +30,19 @@ def armstrong_number(n):
...
+        return True
+
+    return False
 
 
 # In main function user inputs a number to find out if it's an Armstrong or not. Th function armstrong_number is called.
 def main():
     num = int(input("Enter a number to check if it is Armstrong or not: ").strip())
     armstrong_number(num)
+    if armstrong_number(num):
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: We can lose line 41.

Comment (cclauss) [reply maths/armstrong_numbers.py]: `print(f"{num} is {'' if armstrong_number(num) else 'not '}an Armstrong number.")`

**Comment by cclauss** [comment]
Nice work!  Thanks for your contribution.

**Comment by kostogls** [comment]
Thank you!

File: maths/armstrong_numbers.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    # Dividing number into separate digits and find Armstrong number
+    temp = n
+    while temp > 0:
+        rem = temp % 10
+        sum += (rem ** number_of_digits)
+        temp //= 10
+    return n == sum
+
+
+# In main function user inputs a number to find out if it's an Armstrong or not. Th function armstrong_number is called.
```
Comment (cclauss) [inline maths/armstrong_numbers.py]: Can you please add a new pull request that fixes this comment that goes off the screen?

Lines should be limited to 88 characters and the first word in the second sentence is a typo.

Comment (kostogls) [reply maths/armstrong_numbers.py]: Sure!

**Pull request merged**